datacite-draft-doi-handler/src/main/java/no/unit/nva/doi/DraftDoiHandler.java

```diff
@@ -0,0 +1,115 @@
+import no.unit.nva.doi.model.DoiUpdateDto;
+import no.unit.nva.events.handlers.DestinationsEventBridgeEventHandler;
+import no.unit.nva.events.models.AwsEventBridgeDetail;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import no.unit.nva.publication.doi.dto.Publication;
+import no.unit.nva.publication.doi.dto.PublicationHolder;
+import no.unit.nva.transformer.Transformer;
+import no.unit.nva.transformer.dto.DynamoRecordDto;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DraftDoiHandler extends DestinationsEventBridgeEventHandler<PublicationHolder, DoiUpdateDto> {
+
+    public static final String PUBLICATION_IS_MISSING_ERROR = "Publication is missing";
+    public static final String CUSTOMER_ID_IS_MISSING_ERROR = "CustomerId is missing";
+    public static final String TRANSFORMING_PUBLICATION_ERROR = "Error transforming Publication to DataCite XML";
+    private TemporaryDoiClient doiClient;
+
+    private static final Logger logger = LoggerFactory.getLogger(DraftDoiHandler.class);
+
+    /**
+     * Default constructor for DraftDoiHandler.
+     */
+    @JacocoGenerated
+    public  DraftDoiHandler() {
+        this(defaultDoiClient());
+    }
+
+    @JacocoGenerated
+    private static TemporaryDoiClient defaultDoiClient() {
+        // TODO: replace with real DataCite client
+        return new TemporaryDoiClient() {
+            @JacocoGenerated
+            @Override
+            public URI createDoi(String customerId, String metadataDataCiteXml) {
+                return URI.create("http://example.doi");
+            }
+
+            @JacocoGenerated
+            @Override
+            public void updateMetadata(String customerId, String doi, String metadataDataCiteXml) {
+
+            }
+
+            @JacocoGenerated
+            @Override
+            public void setLandingPage(String customerId, String doi, URI url) {
+
+            }
+        };
+    }
+
+    /**
+     * Constructor for DraftDoiHandler.
+     *
+     * @param doiClient doiClient
+     */
+    public DraftDoiHandler(TemporaryDoiClient doiClient) {
+        super(PublicationHolder.class);
+        this.doiClient = doiClient;
+    }
+
+    private DoiUpdateDto createUpdateDoi(Publication input, URI doi) {
+        return new DoiUpdateDto.Builder()
+            .withDoi(doi)
+            .withPublicationId(input.getId())
+            .withModifiedDate(Instant.now())
+            .build();
+    }
+
+    @Override
+    protected DoiUpdateDto processInputPayload(PublicationHolder input,
+                                               AwsEventBridgeEvent<AwsEventBridgeDetail<PublicationHolder>> event,
+                                               Context context) {
+        Publication publication = getPublication(input);
+        URI customerId = getCustomerId(publication);
+        logger.debug("Received request to create draft new DOI for {}", customerId);
```

Comment:
At least exception messages @axthosarouris like to avoid using magic strings, so they are constants. I believe this also counts for logging.